### PR TITLE
Use original filename if include is located in additional include directory

### PR DESCRIPTION
**Pull request by k0ekk0ek**
This PR changes behavior for `idlc` if an included `.idl` file is found in an additional include directory, i.e. a directory specified with the `-I` command line parameter. To achieve this, I've added `IDLC` as a compiler type to `mcpp` (`idlpp` CMake target). If an include is picked up from an additional directory, instead of printing `1` in the extended line marker to signal a new file is entered, it now prints `4` and adds the original filename to the end of the line as a string literal. The directive parser recognizes the `IDLC` specific format and copies the include directive.

~~I still have to add tests~~, but the basic mechanism shouldn't change much (unless there are remarks of course :slightly_smiling_face:).

### CODE REVIEW

**Comment by k0ekk0ek** [comment]
@eboasson, care to have a quick look?

File: src/idl/src/directive.c
Comment (eboasson) [inline src/idl/src/directive.c:160]: Something got mangled here (wrongly indented comment with a meaningless `;` — one would think it existed at some point as an empty loop body)

**Review by eboasson — COMMENTED** [review]
Thanks @k0ekk0ek! A few comments, but nothing major.

File: src/idl/src/directive.c
Comment (k0ekk0ek) [reply src/idl/src/directive.c]: I left it in for now. But yes, it's a bit of a left-over from thinking we might have to signal a system include and since we switched compiler types, that won't ever happen.

File: src/idl/src/directive.c
Comment (k0ekk0ek) [reply src/idl/src/directive.c:139]: The whole loop is a convoluted `basedir` implementation. Now that I fixed the printf, it should make a little more sense.

File: src/idl/src/directive.c
Comment (k0ekk0ek) [reply src/idl/src/directive.c]: Left it in for now. But yes, it's more of a left-over from before switching to a specialized IDLC compiler type. At this point system includes won't happen.

File: src/idl/src/directive.c
Comment (eboasson) [reply src/idl/src/directive.c:139]: I really think you should add a `ddsrt_attribute_format((idl_asprintf, 2, 3));` to its declaration (and the same for all the printf-like functions in the IDL compiler). I went over all the sources once to make sure all such functions are properly annotated, so I either completely forgot about the IDL compiler when doing that or I it did before the merge and missed it in the reviews ...

Either way, I apologise for being late and being insistent about it, but, really, please, just do it!

**Review by eboasson — APPROVED** [review]
Approved, but the printf format annotations need doing. It probably makes the most sense to do that in a separate PR, but I leave that up to you.

**Comment by k0ekk0ek** [comment]
`printf` attributes added where applicable (we can add more in the future), fixed the one bug it unveiled and removed some dead code. merging.

**Pull request merged**
